components/registration-framework/org.wso2.carbon.identity.user.registration.engine/src/main/java/org/wso2/carbon/identity/user/registration/engine/internal/RegistrationFlowEngineServiceComponent.java

```diff
@@ -144,4 +149,44 @@ protected void unsetInputValidationManagementService(InputValidationManagementSe
         LOG.debug("Unsetting the Input Validation Mgt Service in the Registration Flow Engine component.");
         RegistrationFlowEngineDataHolder.getInstance().setInputValidationManagementService(null);
     }
+
+    @Reference(
+            name = "RegistrationExecutionListener",
+            service = RegistrationExecutionListener.class,
+            cardinality = ReferenceCardinality.MULTIPLE,
+            policy = ReferencePolicy.DYNAMIC,
+            unbind = "unsetRegistrationExecutionListener")
+    protected void setRegistrationExecutionListener(RegistrationExecutionListener registrationExecutionListener) {
+
+        LOG.debug("Setting the Registration Execution Listener in the Registration Flow Engine component.");
+        RegistrationFlowEngineDataHolder.getInstance().getRegistrationExecutionListeners()
+                .add(registrationExecutionListener);
+        RegistrationFlowEngineDataHolder.getInstance().getRegistrationExecutionListeners().sort(listenerComparator);
+    }
+
+    protected void unsetRegistrationExecutionListener(RegistrationExecutionListener registrationExecutionListener) {
+
+        LOG.debug("Unsetting the Registration Execution Listener in the Registration Flow Engine component.");
+        RegistrationFlowEngineDataHolder.getInstance().getRegistrationExecutionListeners()
+                .remove(registrationExecutionListener);
+    }
+
+    private static final Comparator<RegistrationExecutionListener> listenerComparator =
```

Comment:
Why we need this?

Reply:
To sort the listeners according to the execution id